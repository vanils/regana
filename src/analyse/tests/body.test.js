
const analyseBody = require('../body.js');
const Scope = require('../../models/Scope');

describe('analyse', () => {
  describe('#analyseBody', () => {
    test('should throw error if trying to analyse not valid entity type', () => {
      const start = 10;
      const end = 400;
      const body = [{ type: 'NotValid' }];
      const scope = new Scope(10, 400, { file: 'File_1' });
      expect(() => analyseBody(start, end, body, scope)).toThrowErrorMatchingSnapshot();
    });
  });
});