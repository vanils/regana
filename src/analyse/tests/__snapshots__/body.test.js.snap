// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analyse #analyseBody should throw error if trying to analyse not valid entity type 1`] = `"Failed to parse, unknown entity type 'NotValid'"`;
